Cron alpha

rascaraficci opened the pull request:
Added some improvements

- Persistence of jobs in database
- Usage of dojot logging library

Commit:
Added support to persist configuration on DB

- Persisted cron job configuration on MongoDB
- Changed the initialization process to load jobs on starting
- Changed to use newer version of dojot-module library
- Added minor bug fixes

Commit:
Changed to use dojot logging library

- Changed how the messages are logged
- Added some bug fixes

Commit:
Fixed some CodeFactor issues

raulnegreiros commented on README.md:
```diff
@@ -1,11 +1,12 @@
 
-- Persist jobs' configuration.
-- Handle cron's restarts.
+- Change to use dojot's logging library
```
Is missing some point? I've seen several uses of dojot's logger during this PR.

rascaraficci replied:
In the commit when I added this TODO item, the cron service wasn't using the logging library. But I forgot to remove it when later I changed to use the library.

raulnegreiros commented on cron.js:
```diff
@@ -26,101 +29,205 @@ class CronManager {
+            this.db.create(tenant, dbEntry).then(() => {
+                this._setCronJob(tenant, jobId, jobSpec).then(() => {
+                    resolve(jobId);
+                });
```
I think is missing a `catch` clause here to treat when the _setCronJob fails.

rascaraficci replied:
You're right!

raulnegreiros commented on db.js:
```diff
@@ -0,0 +1,91 @@
+          });
+    }
+
+    setTenant(tenant) {
```
This name does not sound clear in the class context

rascaraficci replied:
Any suggestion?

rascaraficci replied:
createDatabase(tenant)?

Commit:
Added some fixes

- Added missing 'catch' blocks for some promises
- Updated TODO list

Commit:
Set travis to build/publish docker image

Set travis to build the docker image for the cron
service and publish it to Docker Hub.

raulnegreiros reviewed (approved):
Great
